### PR TITLE
issue#42: Added option of configuring group id during statuspage component creation

### PR DESCRIPTION
**Pull request by usmansharifkhan**
- This PR adds the support for configuring group id when creating statuspage component
- Issue highlighted in https://github.com/yannh/terraform-provider-statuspage/issues/42

### CODE REVIEW

File: statuspage/resource_component.go [hunk 1]
```diff
@@ -22,17 +22,27 @@ func resourceComponentCreate(d *schema.ResourceData, m interface{}) error {
 	onlyShowIfDegraded := d.Get("only_show_if_degraded").(bool)
 	status := d.Get("status").(string)
 	showcase := d.Get("showcase").(bool)
-
-	component, err := sp.CreateComponent(
-		client, d.Get("page_id").(string),
-		&sp.Component{
+	groupId := d.Get("group_id").(string)
```
Comment (smitthakkar96) [inline statuspage/resource_component.go:25]: Would it cause can issue if empty `group_id` is passed? I think the json marshaller would simply omit if it's empty

Comment (usmansharifkhan) [reply statuspage/resource_component.go:25]: Yes @smitthakkar96 , it was causing during testing locally, when empty group_id was passed, so added a if condition below to counter it

Comment (smitthakkar96) [reply statuspage/resource_component.go:25]: do we need a test case for the empty group_id scenario?

File: statuspage/resource_component.go [hunk 2]
```diff
@@ -22,17 +22,27 @@ func resourceComponentCreate(d *schema.ResourceData, m interface{}) error {
...
+	} else {
+		inputComponent = &sp.Component{
+			Name:               &name,
+			Description:        &description,
+			OnlyShowIfDegraded: &onlyShowIfDegraded,
+			Status:             &status,
+			Showcase:           &showcase,
+		}
+	}
+	component, err := sp.CreateComponent(client, d.Get("page_id").(string), inputComponent)
```
Comment (smitthakkar96) [inline statuspage/resource_component.go:45]: nit
``` go
	inputComponent := = &sp.Component{
			Name:               &name,
			Description:        &description,
			OnlyShowIfDegraded: &onlyShowIfDegraded,
			Status:             &status,
			Showcase:           &showcase,
    }
	if groupId != "" {
      inputComponent.GroupID = &groupId
	}
	component, err := sp.CreateComponent(client, d.Get("page_id").(string), inputComponent)

```

Comment (smitthakkar96) [reply statuspage/resource_component.go:45]: Pardon the poor formatting

File: statuspage/resource_component.go (hunk 1 above)
Comment (smitthakkar96) [reply statuspage/resource_component.go:25]: > it was causing issue during testing locally when empty group_id was passed
https://github.com/yannh/terraform-provider-statuspage/pull/43/files#diff-09644ef998d4a5dd5e85566bb231048bcb5e8774b92446d61f1bdc5871b24ca5R195 you tried setting `nil` here?

Comment (smitthakkar96) [reply statuspage/resource_component.go:25]: never mind you will get
```
interface conversion: interface {} is nil, not string
```

**Comment by yannh** [comment]
Hello :wave: And thanks for the contribution :) There are integration tests for which we could add a test case for this to add confidence that it doesn't break anything. If you have a statuspage account, are you able to run those at all?

File: statuspage/resource_component.go (hunk 2 above)
Comment (usmansharifkhan) [reply statuspage/resource_component.go]: done

**Comment by usmansharifkhan** [comment]
Hello @yannh, yes we have a statuspage account. I'll add another test and let know here when its ready for review again, thanks

**Comment by usmansharifkhan** [comment]
Hi @yannh, ready for review again.

**Comment by usmansharifkhan** [comment]
Hi @yannh, your review on this would be great, thanks

**Comment by yannh** [comment]
This broke the tests, had to revert :/
